### PR TITLE
WEBITOOLS-168: Added deep-analysis URL call

### PR DESCRIPTION
**Pull request by leon2Meudon92**

### CODE REVIEW

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java [hunk 1]
```diff
@@ -67,10 +66,9 @@ public Integer processCallCommand() throws Exception {
         }
 
         log.info("Deep-Analysis args:");
-        log.info(String.format("\tApplication: %s%n\tsnapshot name: %s%n\tsleep: %d%n", applicationName, StringUtils.isEmpty(snapshotName) ? "Auto assigned" : snapshotName, sleepDuration));
+        log.info(String.format("\tApplication: %s%n\tsnapshot name: %s%n\tmodule generation type: %s%n\tsleep: %d%n", applicationName, StringUtils.isEmpty(snapshotName) ? "Auto assigned" : snapshotName, moduleGenerationType.toString(), sleepDuration));
```
Comment (AEteve) [inline aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java:69]: This has nothing to do with current ticket, put that in another PR

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java [hunk 2]
```diff
@@ -226,9 +203,12 @@ public String startRunReScanApplication(String applicationGuid, String nodeName,
         if (StringUtils.isNotEmpty(snapshotName)) {
             requestBuilder.snapshotName(snapshotName);
         }
-
+        if (moduleGenerationType != null) {
+            requestBuilder.moduleGenerationType(moduleGenerationType.toString());
```
Comment (AEteve) [inline aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java:207]: another PR needed

**Comment by AEteve** [comment]
This PR needs to be separate into 2 distinct PRs : one for the module type and its correct ticket, and another one for the deep-analysis url call

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java (hunk 1 above)
Comment (leon2Meudon92) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationDeepAnalysisCommand.java:69]: so?

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/jobs/ScanAndReScanApplicationJobRequest.java [hunk 3]
```diff
@@ -21,4 +22,7 @@ public class ScanAndReScanApplicationJobRequest {
     private boolean processImaging = false;
     @Builder.Default
     private boolean publishToEngineering = false;
+
+    @Builder.Default
+    private String moduleGenerationType = ModuleGenerationType.ONE_PER_AU.toString();
```
[the remark this answers is not in the document]
Comment (leon2Meudon92) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/jobs/ScanAndReScanApplicationJobRequest.java:27]: So ? we don't fix issue?

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java (hunk 2 above)
Comment (leon2Meudon92) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java:207]: And what?

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/jobs/ScanAndReScanApplicationJobRequest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (AEteve) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/jobs/ScanAndReScanApplicationJobRequest.java:27]: create another PR with right ticket for that

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java (hunk 2 above)
Comment (AEteve) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/JobsServiceImpl.java:207]: create another PR with right ticket for that
